kuksa_databroker/databroker/src/glob.rs

```diff
@@ -51,3 +52,113 @@ pub fn to_regex(glob: &str) -> Result<Regex, Error> {
     let re = to_regex_string(glob);
     Regex::new(&re).map_err(|_err| Error::RegexError)
 }
+
+fn has_asterisk_in_middle(input: &str) -> bool {
+    let input = input.trim(); // Remove leading and trailing whitespace
+    if input.starts_with('*') || input.ends_with('*') {
+        return false;
+    }
+
+    let asterisk_position = input.find('*');
+    if let Some(pos) = asterisk_position {
+        if input[pos - 1..].contains(".*.") {
+            return true;
+        }
+    }
+
+    false
+}
+
+pub fn to_regex_string_new(glob: &str) -> String {
+    let mut input = String::from(glob);
+
+    //check if asterisk in middle
+    if has_asterisk_in_middle(glob) {
+        input = glob.replace(".*.", ".");
+    }
```

Comment:
Why are we checking if there is an asterisk in the middle?
Why are we replacing `.*.` with `.`. Isn't that just removing the wildcard completely?

Reply:
Yes it is. This is just to cover the special case of 'Vehicle.Cabin.Sunroof.*.Position'.

If we are looking for a single-level child that has a prefix 'Vehicle.Cabin.Sunroof' and ends with 'Position', the regex would be '^Vehicle\\.Cabin\\.Sunroof\\.Position'. The algorithm [below](https://github.com/eclipse/kuksa.val/pull/605/files#diff-f6fc96afb24437d8ff54b4f051da8a6f0b0224f466a088a7e14a188366a8a774R91), which replaces '*' with '[^.]+', does not cover that case. It matches everything in between leading to wrong results, for example: 'Vehicle.Cabin.Sunroof.Shade.Position'

Reply:
I don't get it.

If the input is:
`Vehicle.Cabin.Sunroof.*.Position`

this code will replace it with:
`Vehicle.Cabin.Sunroof.Position`

That will match `Vehicle.Cabin.Sunroof.Position` (which it shouldn't) and not match `Vehicle.Cabin.Sunroof.Shade.Position` (which it should). Right?

Reply:
In the examples [table](https://github.com/boschglobal/kuksa.val/blob/feature/wildcard_matching_doc/doc/wildcard_matching.md) looks like this:
'Vehicle.Cabin.Sunroof.*.Position' returns 'Vehicle.Cabin.Sunroof.Position'

Reply:
Well, that is just wrong (and in turn my mistake) :)